backend/dal/Services/LeaseService.cs

```diff
@@ -1,294 +1,20 @@
-        /// update the tenants on the lease
-        /// </summary>
-        /// <param name="leaseId"></param>
-        /// <param name="pimsPropertyImprovements"></param>
-        /// <returns></returns>
-        public PimsLease UpdateLeaseImprovements(long leaseId, long rowVersion, ICollection<PimsPropertyImprovement> pimsPropertyImprovements)
+        IPimsRepository _pimsRepository;
+        public LeaseService(IPimsRepository pimsRepository)
         {
-            this.User.ThrowIfNotAuthorized(Permissions.LeaseEdit);
-            var existingLease = this.Context.PimsLeases.Include(l => l.PimsPropertyImprovements).Where(l => l.LeaseId == leaseId).AsNoTracking().FirstOrDefault()
-                 ?? throw new KeyNotFoundException();
-            if (existingLease.ConcurrencyControlNumber != rowVersion) throw new DbUpdateConcurrencyException("Unable to save. Please refresh your page and try again");
-
-            this.Context.UpdateChild<PimsLease, long, PimsPropertyImprovement>(l => l.PimsPropertyImprovements, leaseId, pimsPropertyImprovements.ToArray());
-            this.Context.CommitTransaction();
-
-            return Get(existingLease.LeaseId);
+            _pimsRepository = pimsRepository;
         }
-
-        /// <summary>
-        /// update the properties on the lease
-        /// </summary>
-        /// <param name="leaseId"></param>
-        /// <param name="pimsPropertyLeases"></param>
-        /// <returns></returns>
-        public PimsLease UpdatePropertyLeases(long leaseId, long rowVersion, ICollection<PimsPropertyLease> pimsPropertyLeases, bool userOverride = false)
+        public void CheckRowVersion(long leaseId, long rowVersion)
         {
-            this.User.ThrowIfNotAuthorized(Permissions.LeaseEdit);
-            var existingLease = this.Context.PimsLeases.Include(l => l.PimsPropertyLeases).AsNoTracking().FirstOrDefault(l => l.LeaseId == leaseId)
-                 ?? throw new KeyNotFoundException();
-            if (existingLease.ConcurrencyControlNumber != rowVersion) throw new DbUpdateConcurrencyException("Unable to save. Please refresh your page and try again");
-            bool newLeaseProperties = pimsPropertyLeases.Any(p => !existingLease.PimsPropertyLeases.Any(xp => xp.PropertyId == p.PropertyId));
-            existingLease.PimsPropertyLeases = pimsPropertyLeases;
-            var leaseWithAssociatedProperties = AssociatePropertyLeases(existingLease, userOverride, newLeaseProperties);
-
-            this.Context.UpdateChild<PimsLease, long, PimsPropertyLease>(l => l.PimsPropertyLeases, leaseId, leaseWithAssociatedProperties.PimsPropertyLeases.ToArray());
-            this.Context.CommitTransaction();
-
-            return Get(existingLease.LeaseId);
+            PimsLease lease = _pimsRepository.Lease.GetFlat(leaseId, true);
+            if(lease.ConcurrencyControlNumber != rowVersion) throw new DbUpdateConcurrencyException("Unable to save. Please refresh your page and try again");
```

Comment:
It would result in a smaller query if instead of retrieving the entire lease, it would only give you the concurrency control number.